### PR TITLE
Implemented redux actions, reducers

### PR DESCRIPTION
**Pull request by Lana2683**

### CODE REVIEW

File: src/Actions/actions.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    GET_MOVIE,
+    UPDATE_MOVIE,
+    ADD_MOVIE,
+    DELETE_MOVIE,
+    SORT_MOVIES,
+    FILTER_MOVIES,
+} = actionTypes;
+
+export const getMovies = () => async dispatch => {
+    const res = await axios.get('http://localhost:4000/movies')
```
Comment (mmarkelov) [inline src/Actions/actions.js]: i think it's a good idea to move it to constant
```js
const API = 'http://localhost:4000/'
```

File: src/Components/Main.js [hunk 2]
```diff
@@ -1,18 +1,66 @@
...
     return (
-        <>
+        <div className="main">
+            <nav className="nav-bar">
+                <div>
+                    <span className="nav-item" onClick={() => selectAll()}>ALL</span>
+                    <span className="nav-item" onClick={() => filterItem('Documentary')}>DOCUMENTARY</span>
+                    <span className="nav-item" onClick={() => filterItem('Comedy')}>COMEDY</span>
+                    <span className="nav-item" onClick={() => filterItem('Horror')}>HORROR</span>
+                    <span className="nav-item" onClick={() => filterItem('Crime')}>CRIME</span>
```
Comment (mmarkelov) [inline src/Components/Main.js]:
```js
const FILTERS = ['Documentary', 'Comedy', 'Horror', 'Crime']

FILTERS.map(item => <span key={item} className="nav-item" onClick={() => filterItem(item)}>{item.toUpperCase()}</span>)
```

File: src/Components/MovieCard.js [hunk 3]
```diff
@@ -1,37 +1,42 @@
...
      toggleEditMovieModal,
      toggleDeleteMovieModal,
      setMovieInfo,
+     getMovie
     }) => {
     const [isMenuShown, toggleMenu] = useToggle(false);
 
-    const showEditModal = () => {
+    const showEditModal = (id) => {
+        getMovie(id)
```
Comment (mmarkelov) [inline src/Components/MovieCard.js]: `getMovie(id);`

File: src/Components/MovieCard.js [hunk 4]
```diff
@@ -1,37 +1,42 @@
...
         toggleEditMovieModal();
-        toggleMenu()
+        toggleMenu();
     }
 
-    const showDeleteModal = () => {
+    const showDeleteModal = (id) => {
+        getMovie(id)
```
Comment (mmarkelov) [inline src/Components/MovieCard.js]: `getMovie(id);`

File: src/Components/MovieCard.js [hunk 5]
```diff
@@ -54,6 +59,14 @@ MovieCard.prototypes = {
     toggleEditMovieModal: PropTypes.func.isRequired,
     toggleDeleteMovieModal: PropTypes.func.isRequired,
     setMovieInfo: PropTypes.func.isRequired,
+
+    getMovie: PropTypes.func.isRequired,
 }
 
-export default MovieCard;
+const mapStateToProps = () => ({});
```
Comment (mmarkelov) [inline src/Components/MovieCard.js]: cause you don't use it in this component, you can remove it

File: src/Components/MovieModal.js [hunk 6]
```diff
@@ -1,67 +1,165 @@
...
-export const MovieModal = ({ text, isEdit, isDelete, toggleMovieModal }) => (
-    <div className="add-modal">
+const MovieModal = ({ movie, text, isEdit, isDelete, toggleMovieModal, updateMovie, addMovie }) => {
+    const [title, setTitle] = useState('');
+    const [release_date, setRelease_date] = useState('');
+    const [poster_path, setPoster_path] = useState('');
+    const [genres, setGenres] = useState('');
+    const [overview, setOverview] = useState('');
+    const [runtime, setRuntime] = useState('');
+    const [id, setMovieId] = useState(null);
```
Comment (mmarkelov) [inline src/Components/MovieModal.js:15]: I think it's better to use object and change the keys of this object

File: src/Components/MovieModal.js [hunk 7]
```diff
@@ -1,67 +1,165 @@
...
-                        <button className="sbmt-btn border-r">SUBMIT</button>
-                    </>
-                )}
+                    ) : (
+                        <>
+                            <li className={!isEdit ? 'first-input' : ''}>
+                                <TextInput
+                                    name="title"
+                                    title="title"
+                                    type="text"
```
Comment (mmarkelov) [inline src/Components/MovieModal.js]: You don't need to use `type="text"` here, cause it's default value

Comment (mmarkelov) [reply src/Components/MovieModal.js]: Also maybe you can use array to reduce your code?

File: src/Components/TextInput.js [hunk 8]
```diff
@@ -1,16 +1,40 @@
...
+    return (
+        <div className="text-input">
+            <span className="label">{title.toUpperCase()}</span>
+            {type === 'textarea' ? (
+                <textarea
+                    name={name}
+                    placeholder={placeholder}
+                    className="input"
+                    onChange={e => onChange(e.target.value)}
+                    value={value}
```
Comment (mmarkelov) [inline src/Components/TextInput.js:15]: you can move it to common input props:
```js
const inputProps = {
    name,
    value,
    placeholder,
    className: "input",
    onChange: e => onChange(e.target.value)
}

...
{ type === 'textarea' ? <textarea {...inputProps} /> : <input type={type} {...inputProps} />}
```

File: src/Reducers/moviesReducer.js [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+            }
+        }
+
+        case SORT_MOVIES: {
+            const sort = action.payload
+            return {
+                ...state,
+                sortedMoviesList: state.moviesList.sort((a, b) => {
+                    if (sort === 'release_date') {
+                        return Date.parse(new Date(b.release_date).toString()) - Date.parse(new Date(a.release_date).toString());
```
Comment (mmarkelov) [inline src/Reducers/moviesReducer.js:62]:
```js
return new Date(a.release_date) - new Date(b.release_date)
```

File: src/Actions/actions.js (hunk 1 above)
Comment (Lana2683) [reply src/Actions/actions.js]: fixed

**Pull request merged**